src/platform/Linux/NetworkCommissioningWiFiDriver.cpp

```diff
@@ -123,96 +170,247 @@ void LinuxWiFiDriver::Shutdown()
+    memcpy(mStagingNetworks[mCurrentNetId].credentials, credentials.data(), credentials.size());
+    mStagingNetworks[mCurrentNetId].credentialsLen =
+        static_cast<decltype(mStagingNetworks[mCurrentNetId].credentialsLen)>(credentials.size());
 
-    memcpy(mStagingNetwork.ssid, ssid.data(), ssid.size());
-    mStagingNetwork.ssidLen = static_cast<decltype(mStagingNetwork.ssidLen)>(ssid.size());
+    memcpy(mStagingNetworks[mCurrentNetId].ssid, ssid.data(), ssid.size());
+    mStagingNetworks[mCurrentNetId].ssidLen = static_cast<decltype(mStagingNetworks[mCurrentNetId].ssidLen)>(ssid.size());
 
 #if CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
-    mStagingNetwork.networkIdentityLen = 0;
-    mStagingNetwork.clientIdentityLen  = 0;
-    mStagingNetwork.clientIdentityKeypair.reset();
+    mStagingNetworks[mCurrentNetId].networkIdentityLen = 0;
+    mStagingNetworks[mCurrentNetId].clientIdentityLen  = 0;
+    mStagingNetworks[mCurrentNetId].clientIdentityKeypair.reset();
 #endif // CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
+    mCurrentNetId++;
 
     return Status::kSuccess;
 }
 
+void LinuxWiFiDriver::ShiftNetworkAfterRemove()
+{
+    WiFiNetwork tempNetwork[kMaxNetworks];
+    uint8_t count = 0;
+
+    for (uint8_t networkIndex = 0; networkIndex < kMaxNetworks; networkIndex++)
+    {
+        // ssidlen will be 0 if entry is not populated
+        if (!mStagingNetworks[networkIndex].Empty())
+        {
+            tempNetwork[count] = mStagingNetworks[networkIndex];
+            count++;
+        }
+    }
+
+    memcpy(mStagingNetworks, tempNetwork, sizeof(mStagingNetworks));
+}
+
 Status LinuxWiFiDriver::RemoveNetwork(ByteSpan networkId, MutableCharSpan & outDebugText, uint8_t & outNetworkIndex)
 {
     outDebugText.reduce_size(0);
-    outNetworkIndex = 0;
-    VerifyOrReturnError(mStagingNetwork.Matches(networkId), Status::kNetworkIDNotFound);
+    outNetworkIndex   = 0;
+    bool foundNetwork = false;
+
+    VerifyOrReturnError(mCurrentNetId != 0, Status::kBoundsExceeded);
+
+    for (uint8_t networkIndex = 0; networkIndex < kMaxNetworks; networkIndex++)
+    {
+        if (mStagingNetworks[networkIndex].Matches(networkId))
+        {
+            foundNetwork    = true;
+            outNetworkIndex = networkIndex;
+            if (networkIndex == mStagedConnectedNetworkIndex)
+            {
+                mStagedConnectedNetworkIndex = -1;
+            }
+            break;
+        }
+    }
+
+    VerifyOrReturnError(foundNetwork, Status::kNetworkIDNotFound);
 
     // Use empty ssid for representing invalid network
-    mStagingNetwork.ssidLen = 0;
+    mStagingNetworks[outNetworkIndex].ssidLen = 0;
+
+    // Will need to reorder the list.
+    // Shift networks that are at a lower priority than the network to be removed if present in the list.
+    ShiftNetworkAfterRemove();
+    mCurrentNetId--;
+
     return Status::kSuccess;
 }
 
+bool LinuxWiFiDriver::StartReorderingEntries(uint8_t index, int8_t foundNetworkAtIndex)
```

Comment:
I can't tell whether this function is doing what it means to, because I have no idea what it's trying to do.

Reply:
Will add an explanation of the function and whats it tries to achieve.

Reply:
Added the explanation to the NetworkCommissioningDriver.h file